lib/ansible/plugins/strategy/__init__.py

```diff
@@ -541,6 +543,8 @@ def search_handler_blocks_by_name(handler_name, handler_blocks):
                                 for listening_handler_block in iterator._play.handlers:
                                     for listening_handler in listening_handler_block.block:
                                         listeners = getattr(listening_handler, 'listen', []) or []
+                                        if not listeners:
+                                            continue
                                         templar = self.get_handler_templar(listening_handler, iterator)
```

Comment:
> If `listeners` list is not empty, it might also be logical to check if items actually require templating.

https://github.com/ansible/ansible/blob/b8473276455cbeb49629b502f1f163be0c7eb45a/lib/ansible/playbook/base.py#L337
It looks like that `templar` is only used if `attribute` is `class`. `listen` is a list of strings so we might get away with passing `None` as a templar?